### ISSUE TITLE
Mollie Api Loading With composer

### ISSUE THREAD

**Issue by TimVroom** [mollie/Magento#75]
We updated recently to 4.4.x and found the following issue in our setup:

We use composer to update the plugin and this will not symlink the `lib/Mollie` to the api since composer doesn't support git submodules.

Because this link is missing (but the autoloader can be loaded through the composer autoloader). It will throw errors in the checkout (or not showing up at all).

It will throw the "Mollie API not installed" error.

Because the code in `Mollie_Mpm_Helper_Api` is directly checking if its in the lib folder. If its not than it throws the error. (which is our error).

So a solution is to wrap it in an `class_exists` function (which could also throw errors in default magento if i recall). But this basicly means: in the current state a composer only install is not possible.

**Comment by Marvin-Magmodules** [comment]
Thank you for the detailed issue report. We'll look into this next week and push for an update. Maybe drop the submodule all together.

**Comment by Marvin-Magmodules** [comment]
Just removed submodule in [4.4.2-dev](https://github.com/mollie/Magento/tree/4.4.2) and added the PHP API to the package.
Can confirm this as working with composer (magento-hackathon/magento-composer-installer & magento 1.9.3.6). Can you also confirm, so we can release this version?

**Comment by TimVroom** [comment]
This does work for us. Personally im not in favor of adding the php api directly to this package. I think the autoloader isn't needed for magento that is in the lib package. Haven't checked if the namespacing/naming conventions are correct though. If they are then the autoloader isn't really needed.

I think the best solution is to disable the hardcoded checking in the lib folder. This would require some more work to solve it.

So in short: This works correctly, yet a better solution should be viable.

**Comment by Marvin-Magmodules** [comment]
Yes, also not a in favor, but I'm not sure if this will work otherwise. Will do some more investigating, let you know the outcome later this week.

**Comment by Marvin-Magmodules** [comment]
This should do the trick: https://github.com/mollie/Magento/commit/a20b32a263204f06ad212f7502ecbfe9afdd3277

Simple check on class exists. Still need to do some testing on older Magento versions, to see if this is workable as I'm seeings some topic about class_exists throwing errors in Magento (but at this point I'm not able to break this).

**Comment by TimVroom** [comment]
Hello marvin,

This does work for a composer setup, yet it could/should throw warnings or even fatals when installing manually or through modman. Quite a good explanation is here https://www.webguys.de/magento-1/tuerchen-11-the-magento-autoloader-and-external-libraries

Because the class_exists triggers the already defined autoloaders...

It's possible to check if the composer autoloader is know/loaded by having the second argument set to false. Don't have an example at the moment though. 

Must say I like this solution more, but magento is kind of annoying in regards of autoloaders.

**Comment by Frank-Magmodules** [comment]
Hi @TimVroom, 

Sorry for the late reply on this issue. We are still able to install the module using composer, without any issues, so it looks like this issue has been fixed. For future readers, we have setup a wiki page with a step-by-stept guide on how to install the extension using Composer: https://github.com/mollie/Magento/wiki/Install-using-Composer

We are closing this issue now but please feel free to reopen the issue if this still occurs.
